### ISSUE TITLE
Provide a empty ImagePolicy to get the default behavior in 3.9

### ISSUE THREAD

**Issue by nicolaferraro** [minishift/minishift#2379]
### General information

Following https://github.com/openshift/origin/issues/19663. It seems there's a change in behavior in current version of Minishift w.r.t. local lookup of image streams.

The `lookupPolicy` field on ImageStreams is ignored in latest version.

As suggested in the linked issue, the config needs to contain:
```
admissionConfig:
  pluginConfig:
    openshift.io/ImagePolicy: {}
```
In order to get the default behavior.

  * Minishift version: `minishift v1.17.0+f974f0c`
  * OS: Linux
  * Hypervisor: KVM

### Steps to reproduce

  1. Follow the steps in https://github.com/openshift/origin/issues/19663

### Expected
Image is first looked up in imagestreams with `lookupPolicy -> local = true` then on docker hub.

### Actual
Image is only looked up on docker hub.

### Logs
N/A

**Comment by nicolaferraro** [comment]
Comparing the config with --openshift-version `3.9.0` and `3.7.1`. I get the following on `3.9.0`:

```
admissionConfig:
  pluginConfig:
    GenericAdmissionWebhook:
      configuration:
        apiVersion: v1
        disable: false
        kind: DefaultAdmissionConfig
      location: ""
```

i.e. no `openshift.io/ImagePolicy`

While with `3.7.1` I get:
```
admissionConfig:
  pluginConfig:
    GenericAdmissionWebhook:
      configuration:
        apiVersion: v1
        disable: false
        kind: DefaultAdmissionConfig
      location: ""
    openshift.io/ImagePolicy:
      configuration:
        apiVersion: v1
        executionRules:
        - ignoreNamespaceOverride: false
          invertMatch: false
          matchDockerImageLabels: null
          matchImageAnnotations:
          - key: images.openshift.io/deny-execution
            set: false
            value: "true"
          matchImageLabels: null
          matchIntegratedRegistry: false
          matchRegistries: null
          name: execution-denied
          onResources:
          - group: ""
            resource: pods
          - group: ""
            resource: builds
          reject: true
          skipOnResolutionFailure: true
        kind: ImagePolicyConfig
        resolutionRules: null
        resolveImages: ""
      location: ""
```

**Comment by praveenkumar** [comment]
@nicolaferraro I put comment on the original issue https://github.com/openshift/origin/issues/19663#issuecomment-388034450 this is not something we can do. 

You can try using https://docs.openshift.org/latest/minishift/openshift/openshift-client-binary.html#update-openshift-config

**Comment by nicolaferraro** [comment]
Indeed, I was trying to find the point where it is configured but couldn't find it..

**Comment by GrahamDumpleton** [comment]
The command to fix up the config in the interim is:

```
minishift openshift config set --patch '{"admissionConfig": {"pluginConfig": {"openshift.io/ImagePolicy": {"configuration": {"resolutionRules": null, "kind": "ImagePolicyConfig", "executionRules": [{"skipOnResolutionFailure": true, "invertMatch": false, "name": "execution-denied", "onResources": [{"resource": "pods", "group": ""}, {"resource": "builds", "group": ""}], "matchRegistries": null, "matchIntegratedRegistry": false, "matchImageLabels": null, "matchImageAnnotations": [{"set": false, "value": "true", "key": "images.openshift.io/deny-execution"}], "ignoreNamespaceOverride": false, "reject": true, "matchDockerImageLabels": null}], "apiVersion": "v1", "resolveImages": ""}, "location": ""}}}}'
```

**Comment by GrahamDumpleton** [comment]
If using minishift with OKD 3.10, even though image policy configuration exists, image lookup doesn't work at all. Patching the config with what knew used to work doesn't help. So looks like things are totally broken now and no way to fix it. :-(

**Comment by LalatenduMohanty** [comment]
@GrahamDumpleton As discussed in the mail thread about this issue, the root cause needs to be fixed in `oc cluster up`.

**Comment by praveenkumar** [comment]
@GrahamDumpleton looks like as per ML thread discussion this setting should be part of kube-apiserver also, we added `kube-apiserver` patch command also which you want to try and see if that now works and let us know.

Artifact: http://artifacts.ci.centos.org/minishift/minishift/pr/2731/

command: 

```
$ minishift openshift config set --patch '{"admissionConfig": {"pluginConfig": {"openshift.io/ImagePolicy": {"configuration": {"resolutionRules": null, "kind": "ImagePolicyConfig", "executionRules": [{"skipOnResolutionFailure": true, "invertMatch": false, "name": "execution-denied", "onResources": [{"resource": "pods", "group": ""}, {"resource": "builds", "group": ""}], "matchRegistries": null, "matchIntegratedRegistry": false, "matchImageLabels": null, "matchImageAnnotations": [{"set": false, "value": "true", "key": "images.openshift.io/deny-execution"}], "ignoreNamespaceOverride": false, "reject": true, "matchDockerImageLabels": null}], "apiVersion": "v1", "resolveImages": ""}, "location": ""}}}}' --target kube
```

**Comment by gsaslis** [comment]
@praveenkumar just tried the artifact you shared (the macOS build, on an openshfit 3.10 cluster), but no luck I'm afraid... :(

**Comment by GrahamDumpleton** [comment]
It doesn't work in 3.10 for other reasons. Right now from discussions I have seen there will be no fix available in ODK 3.10 version of ``oc cluster up`` for it, which Minishift relies on. Suggest continuing to use minishift v1.22.0+7163416 and OpenShift 3.9 and not upgrade to latest, which uses ODK 3.10 as default.

**Comment by budhram** [comment]
Closing this issue as per @GrahamDumpleton  suggestion as we cannot do much from Minishift side.
